dftimewolf/lib/collectors/gcloud.py

```diff
@@ -55,16 +51,29 @@ def __init__(self,
         state, name=name, critical=critical)
     self.analysis_project = None  # type: gcp_project.GoogleCloudProject
     self.analysis_vm = None  # type: compute.GoogleComputeInstance
-    self.incident_id = str()
     self.remote_project = None  # type: gcp_project.GoogleCloudProject
-    self.remote_instance_name = None  # type: Optional[str]
-    self.disk_names = []  # type: List[str]
-    self.all_disks = False
     self._gcp_label = {}  # type: Dict[str, str]
+    self.__disks_to_copy = []  # type: List[compute.GoogleComputeDisk]
+
+  def _QueueDiskToCopy(self, disk: compute.GoogleComputeDisk) -> None:
+    """Queues a GCE disk to be copied to the analysis VM on Process().
+
+    Args:
+      disk (compute.GoogleComputeDisk): The disk to be queued.
+    """
+    self.__disks_to_copy.append(disk)
+
+  @property
+  def disks_to_copy(self) -> List[compute.GoogleComputeDisk]:
+    """The queued disks that will be copied on Process()."""
+    return self.__disks_to_copy[:]
 
   def Process(self) -> None:
-    """Copies a disk to the analysis project."""
-    for disk in self._FindDisksToCopy():
+    """Copies queued disks to the analysis project."""
+    self.logger.info(
+      'Copying {0:d} disks to {1:s}...'.format(len(self.disks_to_copy),
+                                               self.analysis_vm.name))
+    for disk in self.disks_to_copy:
```

Comment:
Great candidate for migrating to a `ThreadAwareModule` for parallel processing, rather than sequential.